src/status_im/contexts/wallet/sheets/buy_token/view.cljs

```diff
@@ -1,34 +1,64 @@
 (ns status-im.contexts.wallet.sheets.buy-token.view
-  (:require [quo.core :as quo]
+  (:require [oops.core :as oops]
+            [quo.core :as quo]
             [react-native.core :as rn]
             [status-im.contexts.wallet.sheets.buy-token.style :as style]
+            [status-im.feature-flags :as ff]
             [utils.i18n :as i18n]
             [utils.re-frame :as rf]))
 
 (defn- crypto-on-ramp-item
-  [{:keys [name description fees logo-url site-url]}]
-  [quo/settings-item
-   {:title             name
-    :description       :text
-    :description-props {:text description}
-    :tag               :context
-    :tag-props         {:icon    :i/fees
-                        :context fees}
-    :action            :arrow
-    :action-props      {:alignment :flex-start
-                        :icon      :i/external}
-    :image             :icon-avatar
-    :image-props       {:icon logo-url}
-    :on-press          #(rn/open-url site-url)}])
+  [{:keys [name description fees logo-url site-url recurrent-site-url]} _ _ {:keys [tab]}]
+  (let [open-url (rn/use-callback (fn []
+                                    (rn/open-url (if (= tab :recurrent) recurrent-site-url site-url)))
```

Comment:
my vote is for going with dispatch 👍

Reply:
@ilmotta Do you have any notes on this discussion?

I think opening a URL can be a generic event like `:navigate-to` that we can use whenever we want to open a URL in an external browser. I'm tempted to add this to `navigation/events`. wdyt?

Reply:
there is :browser.ui/open-url, although perhaps this can be moved improved upon somehow as this is defied in legacy codebase, probably we should move it? 🤔

Reply:
As @J-Son89 said @ajayesivan, I think there's a strong preference to create events for the UI layer, even though it can be convenient to directly call certain functions. So yes, just like navigation, we would have an event to open any URL. @mmilad75 recently merged a PR that added an *effect* to open URL, the only missing piece is an event.

But since Jamie said `:browser.ui/open-url` already exists, it's something to consider now.

Reply:
@ilmotta if you are talking about this PR https://github.com/status-im/status-mobile/pull/20193, then the effect is triggered from `:browser.ui/open-url`. It's in legacy namespace, I will look into moving it as Jamie suggested.

Reply:
Oh okay, they're tangled, I missed that. Agreed with Jamie's suggestion 👍🏼

Reply:
We are using `:browser.ui/open-url` in multiple places. If you guys don't mind, I will refactor this in a separate PR.